src/frontend/planner_test/tests/testdata/nexmark_source.yaml

```diff
@@ -329,16 +329,20 @@
           |   └─BatchHashAgg { group_key: [window_start, auction], aggs: [count] }
           |     └─BatchHopWindow { time_col: date_time, slide: 00:00:02, size: 00:00:10, output: [auction, window_start] }
           |       └─BatchExchange { order: [], dist: HashShard(auction) }
-          |         └─BatchProject { exprs: [auction, date_time] }
-          |           └─BatchSource { source: "bid", columns: ["auction", "bidder", "price", "channel", "url", "date_time", "extra", "_row_id"], filter: (None, None) }
+          |         └─BatchFilter { predicate: IsNotNull(date_time) }
+          |           └─BatchProject { exprs: [auction, date_time] }
+          |             └─BatchFilter { predicate: IsNotNull(date_time) }
+          |               └─BatchSource { source: "bid", columns: ["auction", "bidder", "price", "channel", "url", "date_time", "extra", "_row_id"], filter: (None, None) }
```

Comment:
@chenzl25 PTAL, is the redundant Filter generated by predicate pushdown across the Share node?

Reply:
Yes, I will separate the batch and stream logical optimization in the future so that plan-sharing wouldn't affect batch query.